runtime/hydradx/src/evm/erc20_currency.rs

```diff
@@ -174,11 +174,25 @@ fn decode_integer(value: Vec<u8>) -> Option<U256> {
 	U256::checked_from(value.as_slice())
 }
 
+fn decode_bool(value: Vec<u8>) -> Option<bool> {
+	if value.len() != 32 {
+		return None;
+	}
+	let mut bytes = [0u8; 32];
+	U256::from(1).to_big_endian(&mut bytes);
+	Some(value == bytes)
+}
+
 fn handle_result(result: CallResult) -> DispatchResult {
 	let (exit_reason, value) = result;
 	match exit_reason {
-		ExitReason::Succeed(ExitSucceed::Returned) => Ok(()),
-		ExitReason::Succeed(ExitSucceed::Stopped) => Ok(()),
+		ExitReason::Succeed(_) => {
+			if Some(false) == decode_bool(value) {
+				Err(DispatchError::Other("evm: erc20 transfer returned false"))
```

Comment:
Could we improve this error message a bit, relating to the domain? When can this happen? Maybe we could hint that.

Reply:
not sure what do you mean. i dont think we can make it more descriptive, this is literally describing this special case when we error out even though transactions succeed.